File: algorithms/2194.cells-in-a-range-on-an-excel-sheet/solution.rb
# @param {String} s
# @return {String[]}
def cells_in_range(s)
  row_start, row_end = s[0], s[3]
  col_start, col_end = s[1], s[4]

  list = []
  for row in row_start..row_end
    for col in col_start.to_i..col_end.to_i
      list << "#{row}#{col}"
    end
  end

  list
end

File: algorithms/2194.cells-in-a-range-on-an-excel-sheet/solution.go
func cellsInRange(s string) []string {
	rowStart, rowEnd := s[0]-64, s[3]-64
	colStart, _ := strconv.Atoi(string(s[1]))
	colEnd, _ := strconv.Atoi(string(s[4]))

	list := []string{}
	for i := rowStart; i <= rowEnd; i++ {
		row := string('A' - 1 + i)
		for j := colStart; j <= colEnd; j++ {
			list = append(list, row+strconv.Itoa(j))
		}
	}

	return list
}
